### DISCUSSION TITLE
專業畫圖python套件推薦

### DISCUSSION THREAD

**Discussion by Bill671** [1101-GS3073-C/Python-Community-GS3073-C#268] [Ideas]
[Highchart](https://www.highcharts.com/)是一個廣為人知的畫圖套件，他提供最視覺化的渲染，
讓**圖形**不像一般的畫圖軟體一樣單調，對於想要做Infographic的朋友來說，是一個非常好的套件，
原始的API 是JS構成，其他衍生出來的像是highchart針對R的客製化則是擁有其最多可調參數，
對於有在用R的朋友是非常友善的。
雖然Python 的 [highchart](https://zhuanlan.zhihu.com/p/33099176) 版本並不如 R版本那麼友善，
但是也足夠應付日常上需求，舉例來說，如果你想畫一個跟Micro m平方一樣專業的股票圖，可以調整時間區段
你可以這麼寫
``` 
from pandas_highcharts.display import display_charts
from pandas_highcharts.core import serialize
import pandas as pd

## 引用數據
df = pd.read_csv('../input/pandas_highchar7842/pandas_highcharts_demo.csv',index_col=0, parse_dates=True)
df = df.sort_index()

## plot
display_charts(df, chart_type="stock", title="德国通货膨胀率")
```
看起來非常炫的圖就會畫出來了 !["股票圖"](https://pic2.zhimg.com/80/v2-68ae83783f4937e70d75acbf21deb35d_1440w.jpg)

當然，他還有很多可調的參數像是figsize、legend、kind等等，這些參數的運用皆跟matplotlib類似，
只是他不是rcParams調整，而是直接在display_charts裡面引用就好，非常推薦給想要做一個專業視覺化網站的人使用

**Comment by enoch-lightning** [comment]
超酷的，我完全沒聽過highchart。你是在做專案的時候有用到highchart嗎？
😺

**Comment by lx3211** [comment]
這對我最近要做的圖表呈現很值得參考，謝謝！

**Comment by BingRong1217** [comment]
是一些很實用的套件

**Comment by limitx0** [comment]
用這個來做圖表好方便

**Comment by Chenszuhong** [comment]
這個主題真是不錯的想法 感覺都很有趣而且很帥 想要知道更多能延生應用的地方

**Comment by Aerofe** [comment]
很實用的套件，大幅降低了工作量

**Comment by awkward-willy** [comment]
感覺也是一個相當實用的套件，或許在未來的專案中是一個可以被考慮的選項之一。不過許多人在 Python 中會以 matplotlib 繪圖，不知道兩者相比有何優缺點？

**Comment by skyling0299** [comment]
這邊有鐵人賽文章~
https://ithelp.ithome.com.tw/articles/10243081
Highcharts 是可以製作複合圖表的，像是柱狀圖和折線圖就常常搭配使用，有這個套件方便許多。

**Comment by Raywei0214** [comment]
感覺很實用，有空來研究看看

**Comment by lukiea** [comment]
我們都很清楚，這是個嚴謹的議題。python對我來說有著舉足輕重的地位，必須要嚴肅認真的看待

**Comment by larry40305** [comment]
每個人都不得不面對這些問題。在面對這種問題時，務必詳細考慮python 繪圖的各種可能。

**Comment by Yilin20021016** [comment]
可惜我的系比較少用到 不然我應該會喜歡上

**Comment by HsuChen88** [comment]
如果在我小時候也有這些教育..我的快樂童年還會不會存在。
不過我還是為我自己很晚接觸程式而感慨。

**Comment by HsuChen88** [comment]
Python 當中比較常用的繪圖套件有這些:
- Matplotlib
- Seaborn
- Plotly
- Bokeh
- Pygal
每個套件有各自的優點，在面對不同情況時可選擇。
